test-utils/src/main/java/com/criteo/publisher/concurrent/TrackingCommandsExecutorWithDelay.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.concurrent;
+
+import androidx.annotation.NonNull;
+import java.util.concurrent.Executor;
+
+/**
+ * Decorator {@link Executor} that adds artifical delay to commands that are run by {@link
+ * TrackingCommandsExecutor}
+ */
+public class TrackingCommandsExecutorWithDelay extends TrackingCommandsExecutor {
```

Comment:
Could you add test on this class ?
Currently I don't think it is tracking task.

In the `execute` method, you're calling `delegate.execute` instead of `super.execute`.

Also, as the delay is done before posting the task, then you're delaying the caller and not the task itself. I guess you want to delay the task.
To achieve this, you'll need to decorate the Runnable itself:

```
val decorated = Runnable {
  Thread.sleep(delay);
  runnable.run();
}
super.execute(decorated);
```

The tracking would start from before the sleep and end after the runnable.

Reply:
My only need was to introduce a delay. Whether it's delaying the task itself or the caller has the same effect. But sure, I can delay the runnable as it better mimicks delay caused by IO.

By the way as far as this version is concerned, it does work correctly (otherwise the tests that rely on this component would have failed). It correctly delegates to the decorated `executor`, by calling: `delegate.execute(runnable)`

Reply:
FWIW, I really wanted to call the delegated object and not `super`. That said, I didn't need to extend from `TrackingCommandExecutor` but just implements the `Executor` interface. This is just decoration and not inheritance.